src/containers/TopicArticlePage/components/TopicArticleTaxonomy.jsx

```diff
@@ -0,0 +1,360 @@
+    this.fetchTaxonomy = this.fetchTaxonomy.bind(this);
+    this.createAndPlaceTopic = this.createAndPlaceTopic.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchTaxonomy();
+  }
+
+  async getSubjectTopics(subjectid) {
+    if (
+      this.state.structure.some(
+        subject => subject.id === subjectid && subject.topics,
+      )
+    ) {
+      return;
+    }
+    try {
+      this.updateSubject(subjectid, { loading: true });
+      const allTopics = await fetchSubjectTopics(subjectid);
+      const groupedTopics = groupTopics(allTopics);
+      this.updateSubject(subjectid, { loading: false, topics: groupedTopics });
+    } catch (e) {
+      handleError(e);
+    }
+  }
+
+  async fetchTaxonomy() {
+    const {
+      article: { language, id },
+    } = this.props;
+    try {
+      this.setState({ status: 'loading' });
+      const [topics, allTopics, allFilters, subjects] = await Promise.all([
+        queryTopics(id, language),
+        fetchTopics(language),
+        fetchFilters(language),
+        fetchSubjects(language),
+      ]);
+
+      const sortedSubjects = subjects
+        .filter(subject => subject.name)
+        .sort(sortByName);
+
+      this.setState(prevState => ({
+        status: 'success',
+        topics,
+        stagedTopicChanges: topics,
+        structure: sortedSubjects,
+        taxonomyChoices: {
+          allTopics: allTopics.filter(topic => topic.name),
+          availableFilters: filterToSubjects(
+            allFilters.filter(filt => filt.name),
+          ),
+          allFilters: allFilters.filter(filt => filt.name),
+        },
+      }));
+    } catch (e) {
+      handleError(e);
+      this.setState({ status: 'error' });
+    }
+  }
+
+  stageTaxonomyChanges({ addTopicId, removeTopicId, path }) {
+    const {
+      article: { title },
+    } = this.props;
+    if (path) {
+      const newTopic = {
+        id: 'staged',
+        name: title,
+        path: `${path}/staged`,
+      };
+      this.setState(prevState => ({
+        stagedTopicChanges: [...prevState.stagedTopicChanges, newTopic],
+      }));
+    } else if (addTopicId) {
+      let newTopic = this.state.taxonomyChoices.allTopics.find(
+        topic => topic.id === addTopicId,
+      );
+      if (!newTopic) {
+        //refresh topics?
+      }
+      this.setState(prevState => ({
+        stagedTopicChanges: [...prevState.stagedTopicChanges, newTopic],
+      }));
+    } else {
+      this.setState(prevState => ({
+        stagedTopicChanges: [
+          ...prevState.stagedTopicChanges.filter(
+            topic => topic.id !== removeTopicId,
+          ),
+        ],
+      }));
+    }
+  }
+
+  async createAndPlaceTopic(topic, articleId) {
+    const newTopicPath = await addTopic({
+      name: topic.name,
+      contentUri: `urn:article:${articleId}`,
+    });
+    const paths = pathToUrnArray(topic.path);
+    console.log(paths);
+    const newTopicId = newTopicPath.split('/').pop();
+    if (paths.length > 2) {
```

Comment:
Kan eventuelt slenge på en kommentar her, eventuelt assigne dette til en variabel med et navn som er forklarende for hvorfor det er > 2?